scl/darwinosl/plugin.conf

```diff
@@ -0,0 +1,81 @@
+#############################################################################
+# Copyright (c) 2023 Hofi <hofione@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+# NOTE: For predicate hints see:
+#						log help predicates
+@define def-osl-predicate "    (eventType == 'logEvent' || eventType == 'lossEvent' || eventType == 'stateEvent' || eventType == 'userActionEvent')
+													  && (logType != 'debug')
+													"
+
+block source darwin-oslog(...) {
+@requires darwinosl "The darwinosl() source driver depends on the darwinosl module"
+	darwinosl(
+		filter-predicate("`def-osl-predicate`")
+		`__VARARGS__`
+		flags(syslog-protocol dont-store-legacy-msghdr assume-utf8)
+	);
+};
+
+block source darwin-oslog-stream(params("--style ndjson --type log --type trace --level info --level debug") user-params("")) {
```

Comment:
I an trying to think that it might be better to have these as a boolean parameter with true and false like:

```suggestion
block source darwin-oslog-stream(
  collect-type-log(true)
  collect-type-trace(true)
  collect-type-activities(false)
  collect-level-info(true)
  collect-level-debug(true)
  collect-level-default(false)
  predicate("")
  process("")
){
............
         program(
             "/usr/bin/log stream --style ndjson $(if ("`collect-type-log`" == "true") "--type log " "") $(if ("`collect-type-trace`" == true) "--type trace " "") $(if ("`collect-type-activities`" == "true") "--type activities " "") $(if ("`collect-level-info`" == "true") "--level info " "") $(if ("`collect-level-debug`" == "true") "--level debug " "") $(if ("`collect-level-default`" == "true") "--level default " "") $(if ("`predicate`" ne "") "--predicate `predicate` " "") $(if ("`process`" ne "") "--predicate `process` " "")"
             flags(
             ......
         )
}
```

A solution like this will make sure that the user will not intentionally set something really bad for themselfs or unset the ` --style ndjson` flag by accident. (I might not get it eight at first but it should work based on the code at https://github.com/syslog-ng/syslog-ng/blob/master/scl/splunk/splunk.conf#L119C22-L119C91)

If the conf file can't do this than it might be worth to do itusing a generator like how https://github.com/syslog-ng/syslog-ng/blob/master/scl/loadbalancer/plugin.conf works

Also this will make it easier to do the documentation for the driver as right now the news.md does not contain the options that can be set (`params` and `user-params` are not referenced in the news file )

Reply:
these 2 here as a kind of an example, and a predefined, default filtered log stream
it is meaningless to modelling all the possible params the user can use here as a filter predicate can be a very-very complex expression
instead, we provide them the filter-predicate that via thay can filter the stream and the static log database as thay want, freely
the forcing of the -type log --type trace --level info --level debug params are intentional, we'd like to have a useful set of retrieved log entries
if it does not for someone thay can even override them, add filter predicate, or completly re-define the stream if needed

Reply:
All the concerns you described are valid, but this is not intended to be a foolproof all in one, supertool
What you described is again would be a lot of effort to implement correctly, so as i wrote above, i would let it to the user now, in a next round we might make an enhancement on this.